pytradfri/color.py

```diff
@@ -0,0 +1,48 @@
+from .const import (ATTR_LIGHT_COLOR_X as X, ATTR_LIGHT_COLOR_Y as Y)
+
+
+KNOWN_XY = {
+  2200: {X: 33135, Y: 27211},
+  2700: {X: 30140, Y: 26909},
+  4000: {X: 24930, Y: 24694},
+}
+
+KNOWN_KELVIN = KNOWN_XY.keys()
+
+KNOWN_X = [v[X] for v in KNOWN_XY.values()]
+
+
+def can_kelvin_to_xy(k):
+    return k is not None and min(KNOWN_KELVIN) <= k <= max(KNOWN_KELVIN)
+
+
+def can_x_to_kelvin(x):
+    return x is not None and min(KNOWN_X) <= x <= max(KNOWN_X)
+
+
+def kelvin_to_xy(k):
+    if k in KNOWN_XY:
+        return KNOWN_XY[k]
+    else:
+        lower_k = max([kk for kk in KNOWN_KELVIN if kk < k])
+        higher_k = min([kk for kk in KNOWN_KELVIN if kk > k])
+        offset = (k - lower_k) / (higher_k - lower_k)
+        lower = KNOWN_XY[lower_k]
+        higher = KNOWN_XY[higher_k]
+        return {
+            coord: int(offset * higher[coord] + (1 - offset) * lower[coord])
+            for coord in [X, Y]
+        }
+
+
+def x_to_kelvin(x):
+    known = next((k for k, v in KNOWN_XY.items() if v[X] == x), None)
+    if known is not None:
```

Comment:
Notice that we check `known`, not `x` and this check is necessary. Of course I can always eliminate this check if I store intermediate list, check its length and return its first element - but it seems to be less clean for me.